admin/services/group.service.js

```diff
@@ -1,11 +1,49 @@
 'use strict'
 
 const groupDataService = require('../services/data-access/group.data.service')
+const pupilDataService = require('../services/data-access/pupil.data.service')
 
 const groupService = {}
 
 groupService.getGroups = async function (query) {
-  return groupDataService.fetchGroups(query)
+  return groupDataService.getGroups(query)
+}
+
+groupService.getPupils = async function (schoolId, groupIdToExclude) {
+  if (!schoolId) { return false }
+  let query = {}
+  let filteredPupil = []
+  let groupedPupils = []
+  let pupils = await pupilDataService.getSortedPupils(schoolId)
+
+  if (groupIdToExclude) {
+    query = { '_id': { $ne: groupIdToExclude } }
+  }
+  const groups = await groupDataService.getGroups(query)
```

Comment:
most of this can be done in SQL when we do the refactor

Reply:
Moving SQL will simplify our code 👍